src/deluge/model/clip/instrument_clip.cpp

```diff
@@ -3216,9 +3268,45 @@ NoteRow* InstrumentClip::getNoteRowFromId(int32_t id) {
 
 bool InstrumentClip::shiftHorizontally(ModelStackWithTimelineCounter* modelStack, int32_t amount) {
 
+	//New community feature as part of Automation Clip View Implementation
+	//If this is enabled, then when you are in a regular Instrument Clip View (Synth, Kit, MIDI, CV), shifting a clip
+	//will only shift the Notes and MPE data (NON MPE automations remain intact).
+
+	//If this is enabled, if you want to shift NON MPE automations, you will enter Automation Clip View and shift the clip there.
+
+	ModelStackWithThreeMainThings* modelStackWithThreeMainThings =
+	    modelStack->addOtherTwoThingsButNoNoteRow(output->toModControllable(), &paramManager);
+
 	if (paramManager.containsAnyParamCollectionsIncludingExpression()) {
-		paramManager.shiftHorizontally(
-		    modelStack->addOtherTwoThingsButNoNoteRow(output->toModControllable(), &paramManager), amount, loopLength);
+		ParamCollectionSummary* summary = paramManager.summaries;
+
+		int32_t i = 0;
+
+		while (summary->paramCollection) {
+
+			ModelStackWithParamCollection* modelStackWithParamCollection =
+			    modelStackWithThreeMainThings->addParamCollection(summary->paramCollection, summary);
+
+			// Special case for MPE only - not even "mono" / Clip-level expression.
+			if (i == paramManager.getExpressionParamSetOffset()) {
+				if (getCurrentUI()
+				    != &automationInstrumentClipView) { //don't shift MPE if you're in the automation view
+					((ExpressionParamSet*)summary->paramCollection)
+					    ->shiftHorizontally(modelStackWithParamCollection, amount, loopLength);
+				}
+			}
+
+			//Normal case
+			else {
+				//this never gets called from Automation View because in the Automation View we shift specific parameters not all parameters
+				if (runtimeFeatureSettings.get(RuntimeFeatureSettingType::AutomationShiftClip)
+				    == RuntimeFeatureStateToggle::Off) {
+					summary->paramCollection->shiftHorizontally(modelStackWithParamCollection, amount, loopLength);
+				}
+			}
+			summary++;
+			i++;
+		}
```

Comment:
This changed logic should maybe go in the `ParamManager::shiftHorizontally` implementation itself, since note rows can also be shifted horizontally on their own (probably only as part of kits so maybe it doesn't matter, but I'm not 100% sure).

Reply:
this type of code is actually found here and in the note row file as well.

I think this might be trickier to implement in the function "void ParamManagerForTimeline::shiftHorizontally" if you can take a look it, please let me know what you think.

I also have a requirement to identify if we're looking at an MPE or Non-MPE parameter.